src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/LoggerMessageDefineAnalyzer.cs

```diff
@@ -0,0 +1,322 @@
+
+        private string? TryGetFormatText(IOperation? argumentExpression)
+        {
+            if (argumentExpression is null)
+                return null;
+
+            switch (argumentExpression)
+            {
+                case ILiteralOperation { ConstantValue: { HasValue: false, Value: string constantValue } }:
+                    return constantValue;
+                case IInterpolatedStringOperation interpolated:
+                    var text = "";
+                    foreach (var interpolatedStringContent in interpolated.Parts)
+                    {
+                        if (interpolatedStringContent is IInterpolatedStringTextOperation textSyntax)
+                        {
+                            text += textSyntax.Text;
+                        }
+                        else
+                        {
+                            return null;
+                        }
+                    }
+                    return text;
+                case INameOfOperation { ConstantValue: { HasValue: true, Value: string constantValue } }:
+                    return constantValue;
+                case INameOfOperation:
+                    // return placeholder from here because actual value is not required for analysis and is hard to get
+                    return "NAMEOF";
+                case IParenthesizedOperation parenthesized:
+                    return TryGetFormatText(parenthesized.Operand);
+                case IBinaryOperation { OperatorKind: BinaryOperatorKind.Add } binary:
+                    var leftText = TryGetFormatText(binary.LeftOperand);
+                    var rightText = TryGetFormatText(binary.RightOperand);
+
+                    if (leftText != null && rightText != null)
+                    {
+                        return leftText + rightText;
+                    }
+
+                    return null;
+                default:
+                    var constant = argumentExpression.ConstantValue;
+                    if (constant.HasValue && constant.Value is string constantString)
+                    {
+                        return constantString;
+                    }
+                    return null;
+            }
+        }
+
+        private static bool FindLogParameters(IMethodSymbol methodSymbol, [NotNullWhen(true)] out IParameterSymbol? message, out IParameterSymbol? arguments)
+        {
+            message = null;
+            arguments = null;
+            foreach (var parameter in methodSymbol.Parameters)
+            {
+                if (parameter.Type.SpecialType == SpecialType.System_String &&
+                    string.Equals(parameter.Name, "message", StringComparison.Ordinal) ||
+                    string.Equals(parameter.Name, "messageFormat", StringComparison.Ordinal) ||
+                    string.Equals(parameter.Name, "formatString", StringComparison.Ordinal))
```

Comment:
Is there a missing parenthesis here or the current form is intended?
`&&` has higher precedence. So a name matching `messageFormat` or `formatString` will meet the condition even if it's not string.